contracts/root/RootERC20Predicate.sol

```diff
@@ -71,22 +75,49 @@ contract RootERC20Predicate is Initializable, IRootERC20Predicate {
      * @inheritdoc IRootERC20Predicate
      */
     function deposit(IERC20Metadata rootToken, uint256 amount) external {
+        uint256 expectedBalance = rootToken.balanceOf(address(this)) + amount;
         _deposit(rootToken, msg.sender, amount);
+        // invariant check to ensure that the root token balance has increased by the amount deposited
+        require((rootToken.balanceOf(address(this)) == expectedBalance), "RootERC20Predicate: UNEXPECTED_BALANCE");
     }
 
     /**
      * @inheritdoc IRootERC20Predicate
      */
     function depositTo(IERC20Metadata rootToken, address receiver, uint256 amount) external {
+        uint256 expectedBalance = rootToken.balanceOf(address(this)) + amount;
         _deposit(rootToken, receiver, amount);
+        // invariant check to ensure that the root token balance has increased by the amount deposited
+        require((rootToken.balanceOf(address(this)) == expectedBalance), "RootERC20Predicate: UNEXPECTED_BALANCE");
+    }
+
+    /**
+     * @inheritdoc IRootERC20Predicate
+     */
+    function depositNativeTo(address receiver) external payable {
```

Comment:
@drinkcoffee primarily for my edification. Do we need to check to ensure the `receiver` address is not a funky address such as `0x0`?

Reply:
This is handled in _deposit